pkg/kubectl/genericclioptions/resource/builder.go

```diff
@@ -704,12 +704,10 @@ func (b *Builder) mappingFor(resourceOrKindArg string) (*meta.RESTMapping, error
 		// if we error out here, it is because we could not match a resource or a kind
 		// for the given argument. To maintain consistency with previous behavior,
 		// announce that a resource type could not be found.
-		// if the error is a URL error, then we had trouble doing discovery, so we should return the original
-		// error since it may help a user diagnose what is actually wrong
-		if _, ok := err.(*url.Error); ok {
-			return nil, err
+		if meta.IsNoMatchError(err) {
+			return nil, fmt.Errorf("the server doesn't have a resource type %q", groupResource.Resource)
 		}
-		return nil, fmt.Errorf("the server doesn't have a resource type %q", groupResource.Resource)
+		return nil, err
```

Comment:
I'm not sure this is the right fix. I think that this message is good if the error is a 404, but we should pass along a different message if it's not a 404.

Can you provide an example of what the output looks like for something like:

`kubectl get foobar`

after this patch is applied?

Reply:
@brendandburns  after this patch is applied,  the output of `kubectl get foobar` would be:
`error: the server doesn't have a resource type "foobar"`